Client/qtTeamTalk/preferencesdlg.cpp

```diff
@@ -1684,17 +1687,29 @@ void PreferencesDlg::slotUpdateTTSTab()
     case TTSENGINE_TOLK :
 #if defined(ENABLE_TOLK)
     {
+        ui.label_ttsoutputmode->show();
+        ui.ttsOutputModeComboBox->show();
         ui.ttsForceSapiChkBox->show();
 
         bool tolkLoaded = Tolk_IsLoaded();
         if (!tolkLoaded)
             Tolk_Load();
         QString currentSR = QString("%1").arg(Tolk_DetectScreenReader());
+        bool hasBraille = Tolk_HasBraille();
+        bool hasSpeech = Tolk_HasSpeech();
         if (!tolkLoaded)
             Tolk_Unload();
         if(currentSR.size())
             ui.ttsForceSapiChkBox->setText(tr("Use SAPI instead of %1 screenreader").arg(currentSR));
         ui.ttsForceSapiChkBox->setChecked(ttSettings->value(SETTINGS_TTS_SAPI, SETTINGS_TTS_SAPI_DEFAULT).toBool());
+        ui.ttsOutputModeComboBox->clear();
+        if (hasSpeech == true && hasBraille == true)
+            ui.ttsOutputModeComboBox->addItem(tr("Speech and Braille"), "SpeechBraille");

```

Comment:
When you use a string as a value for a setting then there's no compile check that the setting is valid. If you use an enum instead you get a compile time check that the setting is valid.

Therefore change the three string values to the following enum:
enum
{
    TTS_OUTPUTMODE_NONE,
    TTS_OUTPUTMODE_SPEECH = 0x1,
    TTS_OUTPUTMODE_BRAILLE = 0x2,
    TTS_OUTPUTMODE_SPEECHBRAILLE = TTS_OUTPUTMODE_SPEECH | TTS_OUTPUTMODE_BRAILLE,
}

When you use a switch-case on an enum you also get a warning if some values are unhandled (which typically is an error).